Alignment/APEEstimation/plugins/createDefaultApeTree.cc

```diff
@@ -0,0 +1,474 @@
+    if(entry%2==1)intervalBegin = *i_id;
+    if(entry%2==0 && id>=intervalBegin && id<*i_id)return true;
+  }
+  return false;
+}
+
+
+
+// ------------ method called to for each event  ------------
+void
+DefaultApeTree::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+  // Same procedure as in ApeEstimatorSummary.cc minus reading of baseline tree
+  
+  // Load APEs from the GT and write them to root files similar to the ones from calculateAPE()
+  edm::ESHandle<AlignmentErrorsExtended> alignmentErrors;
+  iSetup.get<TrackerAlignmentErrorExtendedRcd>().get(alignmentErrors);  
+  
+  // Set up root file for default APE values
+  const std::string defaultFileName(parameterSet_.getParameter<std::string>("resultFile"));
+  TFile* defaultFile = new TFile(defaultFileName.c_str(),"RECREATE");
+  
+  // Naming in the root files has to be iterTreeX to be consistent for the plotting tool
+  TTree* defaultTreeX(nullptr);
+  TTree* defaultTreeY(nullptr);
+  defaultFile->GetObject("iterTreeX;1",defaultTreeX);
+  defaultFile->GetObject("iterTreeY;1",defaultTreeY);
+  // The same for TTree containing the names of the sectors (no additional check, since always handled exactly as defaultTree)
+  TTree* sectorNameTree(nullptr);
+  defaultFile->GetObject("nameTree;1",sectorNameTree);
+  
+  edm::LogInfo("DefaultAPETree")<<"APE Tree is being created";
+  defaultTreeX = new TTree("iterTreeX","Tree for default APE x values from GT");
+  defaultTreeY = new TTree("iterTreeY","Tree for default APE y values from GT");     
+  sectorNameTree = new TTree("nameTree","Tree with names of sectors");
+  
+    
+  // Assign the information stored in the trees to arrays
+  double a_defaultSectorX[16589];
+  double a_defaultSectorY[16589];
+  
+  std::string* a_sectorName[16589];
+  std::map<unsigned int, TrackerSectorStruct>::const_iterator i_sector;
+  for(i_sector = m_tkSector_.begin(); i_sector != m_tkSector_.end(); ++i_sector){
+    const unsigned int iSector(i_sector->first);
+    const bool pixelSector(i_sector->second.isPixel);
+    
+    a_defaultSectorX[iSector] = -99.;
+    a_defaultSectorY[iSector] = -99.;
+    
+    a_sectorName[iSector] = nullptr;
+    std::stringstream ss_sector;
+    std::stringstream ss_sectorSuffixed;
+    ss_sector << "Ape_Sector_" << iSector;
+  
+    ss_sectorSuffixed << ss_sector.str() << "/D";
+    defaultTreeX->Branch(ss_sector.str().c_str(), &a_defaultSectorX[iSector], ss_sectorSuffixed.str().c_str());
+    if(pixelSector){
+      defaultTreeY->Branch(ss_sector.str().c_str(), &a_defaultSectorY[iSector], ss_sectorSuffixed.str().c_str());
+    }
+    sectorNameTree->Branch(ss_sector.str().c_str(), &a_sectorName[iSector], 32000, 00);  
+  }
+  
+  for(std::map<unsigned int,TrackerSectorStruct>::iterator i_sector = m_tkSector_.begin(); i_sector != m_tkSector_.end(); ++i_sector){
+    const std::string& name(i_sector->second.name);      
+    a_sectorName[(*i_sector).first] = new std::string(name);
```

Comment:
this copy can likely be avoided..

Reply:
I changed this into a_sectorName[iSector] = new std::string(i_sector->second.name); and moved it into the upper loop.